src/main/scala/backends/futil/FutilAst.scala

```diff
@@ -0,0 +1,177 @@
+package fuselang.backend.futil
+
+import fuselang.backend.PrettyPrint.Doc
+
+object Futil {
+
+  sealed trait Emitable {
+    def doc(): Doc
+    def emit(): String = this.doc.pretty
+  }
+
+  case class CompVar(name: String) extends Emitable {
+    override def doc(): Doc = Doc.text(name)
+    def port(port: String): CompPort = CompPort(this, port)
+    def addSuffix(suffix: String): CompVar = CompVar(s"$name$suffix")
+  }
+  case class PortDef(id: CompVar, width: Int) extends Emitable {
+    override def doc(): Doc = Doc.parens(id.doc <+> Doc.value(width))
+  }
+
+  /**** definition statements *****/
+  case class Namespace(name: String, comps: List[Component]) extends Emitable {
+    override def doc(): Doc =
+      Doc.parens(
+        Doc.text("define/namespace") <+> Doc.text(name) <@> Doc.nest(
+          Doc.vsep(
+            comps.map(_.doc)
```

Comment:
Here (and below), closing `)` on the same line please. Newlines for `}`, same line for `)`.